bgpd/bgp_attr.c

```diff
@@ -852,11 +853,13 @@ struct attr *bgp_attr_intern(struct attr *attr)
 		else
 			attr->aspath->refcnt++;
 	}
-	if (attr->community) {
-		if (!attr->community->refcnt)
-			attr->community = community_intern(attr->community);
+	if (bgp_attr_get_community(attr)) {
```

Comment:
when we are using bgp_attr_get_community many times at the same place, Guess some thing like this will make it more readable. 

Picked this example from another part of your code changes:
comm = bgp_attr_get_community(attr);
community_unintern(&comm);

Reply:
Will fix.

Reply:
Fixed.